test/common/stats/thread_local_store_test.cc

```diff
@@ -80,12 +80,38 @@ class StatsThreadLocalStoreTest : public testing::Test, public RawStatDataAlloca
     store_->addSink(sink_);
   }
 
-  void setUp() {
+  MOCK_METHOD1(alloc, RawStatData*(const std::string& name));
+  MOCK_METHOD1(free, void(RawStatData& data));
+
+  NiceMock<Event::MockDispatcher> main_thread_dispatcher_;
+  NiceMock<ThreadLocal::MockInstance> tls_;
+  TestAllocator alloc_;
+  MockSink sink_;
+  std::unique_ptr<ThreadLocalStoreImpl> store_;
+};
+
+class HistogramTest : public testing::Test, public RawStatDataAllocator {
+public:
+  HistogramTest() {
+    ON_CALL(*this, alloc(_)).WillByDefault(Invoke([this](const std::string& name) -> RawStatData* {
+      return alloc_.alloc(name);
+    }));
+
+    ON_CALL(*this, free(_)).WillByDefault(Invoke([this](RawStatData& data) -> void {
+      return alloc_.free(data);
+    }));
+
+    EXPECT_CALL(*this, alloc("stats.overflow"));
+    store_.reset(new ThreadLocalStoreImpl(*this));
+    store_->addSink(sink_);
+  }
+
+  void SetUp() override {
```

Comment:
I suspect you want the InSequence object to be declared as a the test class member var, rather than a local member, I confess I don't know what that does.  I'm suggesting that because maybe it enforces some deterministic threading model, but I'm not sure.

Reply:
I am also not sure on this. So just leaving it as is.